File: perl/ase_input_conf_2.pl
#!/usr/bin/perl
use strict;
use YAML::Tiny;

# Automation Scripting Essential Skills 
# Perl

# Input: Reading configuration files
# Technique #2 -- Use YAML::Tiny

our $Yaml;
our $Config;
our $Setting;
our $Value;

if( -f $ARGV[0] )
{
    # Open the config
    $Yaml = YAML::Tiny->read($ARGV[0]);
    # Get the setting
    $Value = $Yaml->[0]->{$ARGV[1]};
    print "Setting = $Value\n";

}
else
{
    print "first arg not a file.\n";
}

File: python/ase_input_conf_2.py
#!/usr/bin/env python

# Automation Scripting Essential Skills 
# Python

# Input: Reading configuration files
# Technique2 -- Use yaml
import sys
import yaml
import pprint

f = open(sys.argv[1])
dataMap = yaml.safe_load(f)
f.close()

#pp = pprint.PrettyPrinter()
#pp.pprint(dataMap)

print dataMap[sys.argv[2]] 
